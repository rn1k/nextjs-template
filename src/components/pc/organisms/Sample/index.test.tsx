import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Sample from './';

describe('Sample', () => {
    describe('スナップショットテスト', () => {
        it('default', () => {
            const tree = renderer.create(
                <Sample isLoading={false} sendRequest={() => {}} />,
            );
            expect(tree).toMatchSnapshot();
        });
    });

    describe('round', () => {
        it('true', () => {
            const tree = renderer.create(
                <Sample
                    isLoading={false}
                    sendRequest={() => {}}
                    round={true}
                />,
            );
            expect(tree).toMatchSnapshot();
        });
    });
});
