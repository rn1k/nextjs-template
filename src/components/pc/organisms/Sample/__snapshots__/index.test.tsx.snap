// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sample round true 1`] = `
Array [
  .c0 {
  display: inline-block;
  padding: 8px 10px;
  margin: 0 auto;
  border-radius: 4px;
  box-sizing: border-box;
  background: #fff;
  border: solid 1px #888;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 8px 24px;
  border-radius: 24px;
}

<a
    className="c0"
    onClick={[Function]}
  >
    <div />
  </a>,
  <div
    onClick={[Function]}
  >
    (React Hooks) ClickCount: 
    0
  </div>,
  .c0 {
  display: inline-block;
  padding-right: 4px;
}

<div
    className="c0"
  >
    <img
      alt=""
      height="30px"
      src="jest-mock-js"
      width="30px"
    />
  </div>,
]
`;

exports[`Sample スナップショットテスト default 1`] = `
Array [
  .c0 {
  display: inline-block;
  padding: 8px 10px;
  margin: 0 auto;
  border-radius: 4px;
  box-sizing: border-box;
  background: #fff;
  border: solid 1px #888;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<a
    className="c0"
    onClick={[Function]}
  >
    <div />
  </a>,
  <div
    onClick={[Function]}
  >
    (React Hooks) ClickCount: 
    0
  </div>,
  .c0 {
  display: inline-block;
  padding-right: 4px;
}

<div
    className="c0"
  >
    <img
      alt=""
      height="30px"
      src="jest-mock-js"
      width="30px"
    />
  </div>,
]
`;
